pkg/apis/infinispan/v1/infinispan_types.go

```diff
@@ -79,6 +79,33 @@ type InfinispanLoggingSpec struct {
 	Categories map[string]string `json:"categories,omitempty"`
 }
 
+// ExposeType describe different exposition methods for Infinispan
+// +kubebuilder:validation:Enum=NodePort;LoadBalancer;Route
+type ExposeType string
+
+const (
+	// ExposeTypeNodePort means a service will be exposed on one port of
+	// every node, in addition to 'ClusterIP' type.
+	ExposeTypeNodePort ExposeType = "NodePort"
+
+	// ExposeTypeLoadBalancer means a service will be exposed via an
+	// external load balancer (if the cloud provider supports it), in addition
+	// to 'NodePort' type.
+	ExposeTypeLoadBalancer ExposeType = "LoadBalancer"
+
+	// ExposeTypeRoute means the service will be exposed via
+	// Route (Openshift resource)
```

Comment:
@rigazilla looks like we need to say that is not only OCP Route, but k8s Ingress also. Or we would like to point in the docs.

Reply:
right @dmvolod, I've changed the commit

Reply:
Thanks, @rigazilla . Looks good.